### PR TITLE
Reduce Number of Warnings in Dynamo.All Solution

### PR DESCRIPTION
**Pull request by JimmySanchezGlobant**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Reduce the number of warnings in Dynamo.All solution, starting from XML comments warnings.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

**Comment by QilongTang** [comment]
Just curious why would adding empty comments help us?

**Comment by JimmySanchezGlobant** [comment]
> Are you adding these manually? I believe the 2 approaches we discussed were:
> 
> 1. Use pragmas to ignore these warnings selectively or on a file basis OR
> 2. Use some automation to generate empty XML docs

@aparajit-pratap Yes, I'm adding these manually, as I could understand pragmas are discarded because the idea is not to ignore these warnings, and automation is discarded because @sm6srw suggested not modifying the XAML files and basically focus on the classes.

**Comment by aparajit-pratap** [comment]
> > Are you adding these manually? I believe the 2 approaches we discussed were:
> > 
> > 1. Use pragmas to ignore these warnings selectively or on a file basis OR
> > 2. Use some automation to generate empty XML docs
> 
> @aparajit-pratap Yes, I'm adding these manually, as I could understand pragmas are discarded because the idea is not to ignore these warnings, and automation is discarded because @sm6srw suggested not modifying the XAML files and basically focus on the classes.

I disagree, **adding these manually is going to be very time-consuming**. pragmas if added conditionally (selectively), not universally, will not be a blanket ignoring of all such warnings. I'm not familiar with @sm6srw's approach, but I feel it's worth exploring his approach as well as a way to avoid doing this manually.

**Comment by pinzart90** [comment]
I am hoping we can suppress all warnings on *.xml files (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/suppress-warnings) and run automation on *.cs files to generate boilerplate documentation (hopefully not just empty comments, maybe the class/member name)

**Comment by JimmySanchezGlobant** [comment]
Thank you all for your kind suggestions, I was investigating about automate the generation of the XML documentation but I found only VS extension called GhostDoc that is commercial and it allows only conversion method by method in the free version (which could be similar to the manual option), because of that I discarded this approach, besides, I investigated warnings suppression and I decided to modify just the .csproj files adding the CS1591 warning to the specific list of warnings to be suppressed for the projects that presented that warning (this is reflected on the last commit), practically removing almost 4000 warnings, so, just please let me know if my approach is correct and if isn't I'm open hearing other alternatives.

**Comment by sm6srw** [comment]
My thinking was that adding boiler plate comments (automatically with some tool) would allow us to still get warnings for any new methods added in a file/class. I did not get that far when I looked at this during recharge but GhostDoc was one tool others had used in cases like this. I did not realize that the full version requires a license.

**Comment by sm6srw** [comment]
So with that said: A selective suppression using pragmas is probably best but we need to decide what warnings we actually want to eventually fix and then adapt the suppression for each warning based on that.

**Comment by JimmySanchezGlobant** [comment]
I know that the best approach is just to add all the non-critical suppressed warnings to the CS_SDK.props file but for any strange reason when I build the solution it still shows me all these warnings, so, I had to modify the .csproj files for all projects that presented these warnings, then, these changes lowered the number of warnings from 6000 to 500 approx., just let me know if my approach is right and if it's not I'm open to hearing better alternatives.

File: src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.csproj [hunk 1]
```diff
@@ -9,6 +9,30 @@
     <RootNamespace>Dynamo.DocumentationBrowser</RootNamespace>
     <AssemblyName>DocumentationBrowserViewExtension</AssemblyName>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (pinzart90) [inline src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.csproj:12]: why do we need all these conditions ?
Can't we just add a single PropertyGroup for all config/platform configs (i.e no condition) ?

ex.
```
<PropertyGroup>
<NoWarn>MSB3539;CS1591;NUnit2005;NUnit2007;CS0618;CS0612;CS0672</NoWarn>
  </PropertyGroup>
```

File: src/Config/CS_SDK.props [hunk 2]
```diff
@@ -19,7 +19,7 @@
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <TestProjectDefaults Condition="$(MSBuildProjectDirectory.Contains('test'))">true</TestProjectDefaults>
     <TestProjectDefaults Condition=" '$(TestProjectDefaults)' == '' ">false</TestProjectDefaults>
-    <NoWarn>MSB3539</NoWarn><!--The value of the property "BaseIntermediateOutputPath" was modified after it was used by MSBuild-->
+    <NoWarn>MSB3539;CS1591;NUnit2005;NUnit2007;CS0618;CS0612;CS0672</NoWarn><!--The value of the property "BaseIntermediateOutputPath" was modified after it was used by MSBuild-->
```
Comment (aparajit-pratap) [inline src/Config/CS_SDK.props:22]: Can you add comments for each of these warning codes explaining what they're for?

Comment (JimmySanchezGlobant) [reply src/Config/CS_SDK.props]: Done.

**Review by pinzart90 — APPROVED** [review]
LGTM.
Would have been nice to only put the warnings in a single file though. We will need to place them in all .csproj files (even if they are not yet in the dotnet6 solution)

**Review by aparajit-pratap — APPROVED** [review]

**Comment by aparajit-pratap** [comment]
@JimmySanchezGlobant any idea what's up with these failures: 
image

Are they failing in .NET6?

**Comment by JimmySanchezGlobant** [comment]
> @JimmySanchezGlobant any idea what's up with these failures: image
> 
> Are they failing in .NET6?

@aparajit-pratap No idea, the changes I've made are just related with suppressing warnings, I didn't change any kind of logic neither tests.

**Pull request merged**